geoh5py/ui_json/enforcers.py

```diff
@@ -314,24 +315,16 @@ def enforce(self, name: str, value: Any):
         if not self.rule(value):
             raise self.validation_error(
                 name,
-                [k for k in self.validations if k not in value],
+                [k for k in self.validations if k not in self.collection(value)],
             )
 
     def rule(self, value: Any) -> bool:
         """True if all required parameters are in 'value' collection."""
-        return all(k in value for k in self.validations)
-
-    @property
-    def validations(self) -> list[str]:
-        if self._validations is None:
-            self._validations = []
-        return self._validations
+        return all(k in self.collection(value) for k in self.validations)

```

Comment:
Can we get away without the collection? Seems backward compared to the other Enforcers, where the `rule` takes in the value to check. Here the value is used to validate? 

```suggestion
        return all(k in self.validations for k in value)

```

Reply:
The suggestion doesn't work since value is the full dictionary of uijson parameter/values.